api/arceos_posix_api/src/imp/pthread/specific.rs

```diff
@@ -0,0 +1,44 @@
+/* Copyright (c) [2023] [Syswonder Community]
+ *   [Rukos] is licensed under Mulan PSL v2.
+ *   You can use this software according to the terms and conditions of the Mulan PSL v2.
+ *   You may obtain a copy of Mulan PSL v2 at:
+ *               http://license.coscl.org.cn/MulanPSL2
+ *   THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT, MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
+ *   See the Mulan PSL v2 for more details.
+ */
+
+use crate::ctypes;
+use axtask::specific::DestrFunction;
+use core::ffi::{c_int, c_void};
+
+/// Allocate a specific key for a process shared by all threads.
+pub unsafe fn sys_pthread_key_create(
+    key: *mut ctypes::pthread_key_t,
+    destr_function: Option<DestrFunction>,
+) -> c_int {
+    if let Some(k) = axtask::current().alloc(destr_function) {
+        unsafe {
+            *key = k as ctypes::pthread_key_t;
+        }
+        0
+    } else {
+        -1
```

Comment:
Should not return -1 directly.
Return EAGAIN if running out of keys.
And wrap functions with `syscall_body`